backend/src/auth/auth.service.ts

```diff
@@ -200,4 +203,86 @@ export class AuthService {
       )
     }
   }
+
+  /**
+   * @description Send password reset email
+   */
+
+  async startForgottenPasswordFlow(email: string) {
+    const user = await this.userService.findOneByEmail(email)
+
+    if (!user) {
+      // No error to prevent mail checking
+      return
+    }
+
+    // Default to case that user does not have a verified email i.e. they have to contact support
+    let mailContent: MailTemplateContent = {}
+    let mailTemplate = MailTemplates.PASSWORD_RESET_SUPPORT
+
+    if (user.hasVerifiedEmail) {
+      const resetToken = this.jwtService.sign(
+        {
+          id: user._id,
+        } as PasswordResetJWTPayload,
+        {
+          secret: this.configService.get('JWT_PASSWORD_RESET_SECRET'),
+          expiresIn: this.configService.get('JWT_PASSWORD_RESET_EXPIRE_TIME'),
+        },
+      )
+      const resetUrl = `${this.configService.get(
+        'BACKEND_DOMAIN',
+      )}/auth/verify-email?token=${resetToken}`
+
+      mailContent = {
+        resetUrl,
+      } as PasswordResetMailData
+      mailTemplate = MailTemplates.PASSWORD_RESET
+    }
+
+    const mailData: MailData = {
+      content: {
+        subject: 'Email reset',
+        templateContent: mailContent,
+        contentTemplate: mailTemplate,
+      },
+      recipient: {
+        recipient: email,
+      },
+    }
+
+    try {
+      await this.mailService.scheduleMailNow(mailData)
+    } catch (error) {
+      this.logger.warn(
+        `Could not send password reset email due to an error ${error}`,
+      )
+      throw new ServiceUnavailableException(
+        'We cannot send emails right now, please try again later',
+      )
+    }
+  }
```

Comment:
this just catches and throws the underlying exception again, is it necessary to have this jsut for the logger.warn?

Especially the duplicate message, would be enough to just throw error since its basically the same message